src/python_testing/TC_FAN_4_1.py

```diff
@@ -0,0 +1,323 @@
+                self.skip_step(step_num + 1)
+                self.skip_step(step_num + 2)
+                self.skip_step(step_num + 3)
+                self.skip_step(step_num + 4)
+                self.skip_step(step_num + 5)
+                self.skip_step(step_num + 6)
+                step_num += num_substeps
+                return
+
+            self.step(step_num + 1)
+            fan_mode = await self.read_single_attribute_check_success(cluster=fan, attribute=fan.Attributes.FanMode)
+            asserts.assert_equal(fan_mode, expected_mode, "Incorrect FanMode")
+
+            self.step(step_num + 2)
+            percent_setting = await self.read_single_attribute_check_success(cluster=fan, attribute=fan.Attributes.PercentSetting)
+            if expected_percent_setting is not None:
+                asserts.assert_equal(percent_setting, expected_percent_setting, "Incorrect percent setting")
+            else:
+                asserts.assert_not_equal(percent_setting, 0, "Incorrect percent setting")
+
+            self.step(step_num + 3)
+            logging.info(f"Waiting for {wait_s} seconds to give the fan a chance to respond")
+            time.sleep(wait_s)
+
+            self.step(step_num + 4)
+            percent_current = await self.read_single_attribute_check_success(cluster=fan, attribute=fan.Attributes.PercentCurrent)
+            asserts.assert_equal(percent_current, 0, "Fan turned on while On/Off cluster was set to off")
+
+            if has_spd:
+                self.step(step_num + 5)
+                speed_setting = await self.read_single_attribute_check_success(cluster=fan, attribute=fan.Attributes.SpeedSetting)
+                if expected_speed_setting is not None:
+                    asserts.assert_equal(speed_setting, expected_speed_setting,
+                                         "Speed was not adjusted when percent was adjusted while fan was off")
+                else:
+                    asserts.assert_not_equal(speed_setting, 0, "Incorrect speed setting")
+
+                self.step(step_num + 6)
+                speed_current = await self.read_single_attribute_check_success(cluster=fan, attribute=fan.Attributes.SpeedCurrent)
+                asserts.assert_equal(speed_current, 0, "Current speed is not 0 even though the fan is off")
+            else:
+                self.skip_step(step_num + 5)
+                self.skip_step(step_num + 6)
+            step_num += num_substeps
+
+        lowest_mode = self.get_fan_modes(supported_fan_modes)[1]
+        await verify_onoff_off(fan.Attributes.PercentSetting(1), lowest_mode, 1, 1)
+        await verify_onoff_off(fan.Attributes.PercentSetting(0), fan.Enums.FanModeEnum.kOff, 0, 0)
+        await verify_onoff_off(fan.Attributes.FanMode(fan.Enums.FanModeEnum.kHigh), fan.Enums.FanModeEnum.kHigh, None, None)
+        await verify_onoff_off(fan.Attributes.FanMode(fan.Enums.FanModeEnum.kOff), fan.Enums.FanModeEnum.kOff, 0, 0)
```

Comment:
Use keyword args here too, to make these many-arg calls easier to follow.

Reply:
done.